test/integration/visual_shininess.cc

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <mutex>
+#include <functional>
+
+#include <ignition/math/Rand.hh>
+
+#include "test_config.h"
+
+#include "gazebo/physics/physics.hh"
+#include "gazebo/sensors/sensors.hh"
+#include "gazebo/common/Timer.hh"
+#include "gazebo/rendering/Camera.hh"
+#include "gazebo/sensors/CameraSensor.hh"
```

Comment:
nit: other than `test_config.h` and `gazebo/common/Timer.hh`, are these imports needed?

Reply:
many weren't need, so I removed them in https://github.com/osrf/gazebo/pull/3231/commits/36f70c3af1045710d6463f621e6e422c26a192b4, though I also added `gazebo/rendering/rendering.hh`